docs/source/extension/extension_migration.rst

```diff
@@ -6,6 +6,21 @@
 Extension Migration Guide
 ================================================
 
+
+JupyterLab 4.1 to 4.2
+---------------------
+
+API updates
+^^^^^^^^^^^
+
+- The ``CodeEditor.ICoordinate`` interface was corrected to not include ``toJSON()``, ``x``, ``y``,
+  ``width`` and ``height``; these properties were never set by methods returning ``ICoordinate``
+  and they were never used by methods accepting it.
+- ``CodeEditor.getCoordinateForPosition`` return type was corrected to clarify that it can return
```

Comment:
Should this be considered as a breaking change? Thinking this may affect some extensions not checking for the value to be `null`.

Reply:
Or I guess this change is ok, since it's a "fix" of the API.

Reply:
Yes, this is a bug fix - if an extension was not checking for a non-null value (which it most likely did not) its users would have a runtime error in some cases which is arguably worse than the developer having a build time error after upgrading packages.